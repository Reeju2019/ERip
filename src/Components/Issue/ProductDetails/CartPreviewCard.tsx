interface ICartPreviewCardProps {
  item: {
    model?: string
    serviceImage: string
    serviceName: string
    price: number
    off: number
    repairTime: string
    warranty: string
    repairType: string
  }
}

const CartPreviewCard: React.FunctionComponent<ICartPreviewCardProps> = (props) => {
  const item = props
  return (
    <>
      <div className='my-3 d-flex flex-row justify-content-between text-muted lh-1'>
        <p className='m-0'>
          {item.item.model} - {item.item.serviceName}
        </p>
        <p className='m-0 text-primary'>â‚¹ {item.item.price}</p>
      </div>
      <hr className='m-0' />
    </>
  )
}

export default CartPreviewCard
