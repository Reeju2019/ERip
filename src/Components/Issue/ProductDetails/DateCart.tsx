import * as React from 'react'

interface IDateCardProps {
  date: any[]
  setActiveDate: React.Dispatch<React.SetStateAction<any[]>>
}

const DateCard: React.FunctionComponent<IDateCardProps> = ({ date, setActiveDate }) => {
  const handleClick = () => {
    setActiveDate(date)
  }
  return (
    <>
      <div onClick={handleClick} className=''>
        <h6>{date[0]}</h6>
        <h4>{date[1]}</h4>
        <h6>{date[2]}</h6>
      </div>
    </>
  )
}
export default DateCard
