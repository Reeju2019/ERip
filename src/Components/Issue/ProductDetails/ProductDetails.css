.prodImg {
  width: 10vmax;
  margin: 2vmax;
}

.servicesImg {
  width: 100%;
  display: block;
  margin: auto;
}

.prodRow {
  width: 100%;
}

/* service card */
.serviceParagraph {
  font-size: 0.8vmax;
  font-weight: bold;
  color: gray;
}

.lineHeight {
  line-height: 1;
}
.sticThrough {
  text-decoration: line-through;
  color: gray;
  font-size: 1.2vmax;
}
.colorGreen {
  color: green;
  font-weight: bold;
  font-size: 1vmax;
}

.serviceBtn {
  border: 1px solid blue;
  padding: 4px;
  width: 5.5vmax;
  font-size: 0.9vmax;
  border-radius: 5px;
}

.knowMore {
  font-size: 1vmax;
  margin-left: 18%;
  font-weight: bold;
  color: blue;
}

.knowMorePara {
  cursor: pointer;
  width: 40vw;
}
.btnbtn {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.removeBtn {
  background-color: blue;
  color: white;
}

.offerBorder {
  border: 1px solid blue;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  cursor: pointer;
}

.couponText {
  color: blue;
  font-size: 15px;
  margin-left: 3%;
  margin-top: 2%;
  line-height: 1;
}

.fontAwesome {
  color: blue;
  align-items: center;
  align-self: center;
  margin-right: 3%;
}

.applyBtn {
  font-size: 15px;
  /* min-width: 2vmax; */
}

.slot_card_body {
  margin: 10px;
  padding: 20px;
}
.radio_slot {
  height: 80px;
  width: 80px;
}
