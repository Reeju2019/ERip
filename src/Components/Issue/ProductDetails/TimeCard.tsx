import * as React from 'react'

interface ITimeCardProps {
  time: string
  setActiveTime: React.Dispatch<React.SetStateAction<string>>
}

const TimeCard: React.FunctionComponent<ITimeCardProps> = ({ time, setActiveTime }) => {
  const TimeClicked = () => {
    setActiveTime(time)
  }
  return (
    <>
      <div onClick={TimeClicked}>{time}</div>
    </>
  )
}

export default TimeCard
