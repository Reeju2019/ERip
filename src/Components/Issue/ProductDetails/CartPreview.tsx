import { Button, Col, Container, Modal, Row } from 'react-bootstrap'
import CartPreviewCard from './CartPreviewCard'
import IssueData from '../../../Data/IssuePage.mock.json'
import { useEffect, useState } from 'react'
import Slider from 'react-slick'
import TimeCard from './TimeCard'
import DateCard from './DateCart'
import { BiArrowBack } from 'react-icons/bi';
// import moment from 'moment'
// import timeSlotData from '../../../Data/Timeslot.data.json'

interface ICartPreview {
  cartItem:
    | {
        model?: string
        serviceImage: string
        service_name: string
        price: number
        off: number
        repairTime: string
        warranty: string
        repairType: string
      }[]
    | undefined
}

// interface FilterTimeSlot {
//   id: number
//   start_time_hour: number
//   start_time: string
//   end_time: string
// }

const CartPreview: React.FunctionComponent<ICartPreview> = (props) => {
  const cartItem = props
  const [show3, setShow3] = useState(false)
  const handleClose3 = () => setShow3(false)
  const handleShow3 = () => setShow3(true)
  const [slot, setSlot] = useState<string[][]>([])
  const [slotTime, setSlotTime] = useState<string[]>([])
  const [activeTime, setActiveTime] = useState('')
  const [activeDate, setActiveDate] = useState<string[]>([])
  // console.log(activeDate, activeTime)

  const totalDiscount = (allItem: ICartPreview) => {
    let discount = 0
    allItem.cartItem?.map((item) => {
      const offer = (item.price * item.off) / 100
      discount += offer
      Math.ceil(offer)
    })
    return Math.ceil(discount)
  }
  const totalAmount = (allItem: ICartPreview) => {
    let total = 0
    allItem.cartItem?.map((item) => {
      total += item.price
    })
    return total
  }
  const currentDateString = (i: number, currentDate: number) => {
    if (currentDate + i === 1 || (i + currentDate > 20 && (i + currentDate) % 10 === 1)) {
      return String(currentDate + i) + 'st'
    } else if (currentDate + i === 2 || (i + currentDate > 20 && (i + currentDate) % 10 === 2)) {
      return String(currentDate + i) + 'nd'
    } else if (currentDate + i === 3 || (i + currentDate > 20 && (i + currentDate) % 10 === 3)) {
      return String(currentDate + i) + 'rd'
    } else {
      return String(currentDate + i) + 'th'
    }
  }

  useEffect(() => {
    const Slot = () => {
      const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
      const months = [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December',
      ]
      const currentTimestamp = new Date()
      const AvailableSlotList = []
      for (let i = 0; i < 8; i++) {
        const currentDay = currentTimestamp.getDay()
        const currentMonth = currentTimestamp.getMonth()
        const currentDate = currentTimestamp.getDate()
        AvailableSlotList.push([
          currentDay + i > 6 ? days[currentDay + i - 7] : days[currentDay + i],
          currentDateString(i, currentDate),
          i > 1 ? months[currentMonth] : i === 0 ? 'Today' : 'Tomorrow',
        ])
      }
      // console.log(AvailableSlotList)
      return AvailableSlotList
    }
    const time = () => {
      const times = []
      const startHours = 9
      const endHours = 9 + 2
      for (let i = 0; i < 6; i++) {
        let sHours = startHours + i * 2
        let eHours = endHours + i * 2
        const startAmpm = sHours > 11 ? 'PM' : 'AM'
        if (sHours > 12) {
          sHours -= 12
        }
        const endAmpm = eHours > 11 ? 'PM' : 'AM'
        if (eHours > 12) {
          eHours -= 12
        }
        const strTime =
          sHours.toString() + ' ' + startAmpm + '-' + eHours.toString() + ' ' + endAmpm
        times.push(strTime)
      }
      return times
    }
    time()
    setSlot(Slot())
    setSlotTime(time())
  }, [])

  const SlickArrowLeft = ({ currentSlide, ...props }: any) => (
    <button
      {...props}
      className={'slick-prev slick-arrow' + (currentSlide === 0 ? ' slick-disabled' : '')}
      aria-hidden='true'
      aria-disabled={currentSlide === 0 ? true : false}
      type='button'
    >
      Previous
    </button>
  )
  const SlickArrowRight = ({ currentSlide, slideCount, ...props }: any) => (
    <button
      {...props}
      className={
        'slick-next slick-arrow' + (currentSlide === slideCount - 1 ? ' slick-disabled' : '')
      }
      aria-hidden='true'
      aria-disabled={currentSlide === slideCount - 1 ? true : false}
      type='button'
    >
      Next
    </button>
  )
  const settings = {
    accessibility: true,
    adaptiveHeight: false,
    arrows: true,
    autoplay: false,
    autoplaySpeed: 3000,
    centerMode: false,
    centerPadding: '50px',
    cssEase: 'ease',
    dots: false,
    dotsClass: 'slick-dots',
    draggable: true,
    easing: 'linear',
    edgeFriction: 0.15,
    fade: false,
    focusOnSelect: false,
    focusOnChange: false,
    infinite: false,
    initialSlide: 0,
    mobileFirst: false,
    nextArrow: <SlickArrowRight />,
    pauseOnDotsHover: false,
    pauseOnFocus: true,
    pauseOnHover: true,
    prevArrow: <SlickArrowLeft />,
    respondTo: 'window',
    speed: 500,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
    rows: 1,
    rtl: false,
    slide: '',
    slidesPerRow: 1,
    slidesToScroll: 1,
    slidesToShow: 4,
    swipe: true,
    swipeToSlide: false,
    touchMove: true,
    touchThreshold: 5,
    useCSS: true,
    useTransform: true,
    variableWidth: false,
    vertical: false,
    verticalSwiping: false,
    waitForAnimate: true,
    zIndex: 1000,
  }

  return (
    <>
      <Container fluid className='w-100 mx-1'>
        <Row>
          <Col className='bg-white mt-0 p-5'>
            <h5>{IssueData.constData.cartPreview.heading}</h5>
            <hr />
            <div>
              {cartItem &&
                cartItem.cartItem?.map((item, index) => {
                  return (
                    <div key={index}>
                      <CartPreviewCard item={item} />
                    </div>
                  )
                })}
            </div>
            <div>
              <div className='my-3 d-flex flex-row justify-content-between lh-1'>
                <p className='m-0'>
                  {IssueData.constData.cartPreview.youSaved}
                  <span className='text-success'>
                    {' '}
                    (
                    {Math.floor((totalDiscount(cartItem) * 100) / totalAmount(cartItem))
                      ? Math.floor((totalDiscount(cartItem) * 100) / totalAmount(cartItem))
                      : 0}
                  </span>
                  <span className='text-success'>%)</span>
                </p>
                <p className='m-0 text-primary'>₹ {totalDiscount(cartItem)}</p>
              </div>
              <hr className='m-0' />
            </div>
            <div>
              <div className='my-3 d-flex flex-row justify-content-between lh-1'>
                <p className='m-0'>{IssueData.constData.cartPreview.totalAmount}</p>
                <p className='m-0 text-primary'>
                  ₹ {totalAmount(cartItem) - totalDiscount(cartItem)}
                </p>
              </div>
              <hr className='m-0' />
            </div>

            <Row className='px-2 mt-5'>
              <Button variant='primary' onClick={handleShow3}>
                {IssueData.constData.cartPreview.bookNow}
              </Button>
              <Modal
                show={show3}
                onHide={handleClose3}
                keyboard={false}
                {...props}
                aria-labelledby='example-custom-modal-styling-title'
                centered
                style={{ textAlign: 'center' }}
                backdrop='static'
              >
                <Modal.Header className='appoinmentModal'>
                <button className='book_modal_back' onClick={handleClose3}><BiArrowBack className='closeButton'/></button>
                  <Modal.Title id='example-custom-modal-styling-title'>
                    Schedule Appointment
                  </Modal.Title>
                </Modal.Header>
                <Modal.Body className='appoinmentModal'>
                  <Container>
                    <div>
                      <p className='m-0'>Select Date</p>
                      <Slider {...settings}>
                        {slot.map((item, index) => {
                          return (
                            <div
                              key={index}
                              className={
                                'col-3 p-2 m-3 fw-bold dateCard ' +
                                (activeDate === item ? 'active' : 'times')
                              }
                            >
                              <DateCard date={item} setActiveDate={setActiveDate} />
                            </div>
                          )
                        })}
                      </Slider>
                    </div>
                  </Container>
                </Modal.Body>
                <Modal.Footer>
                  <div>
                    <p className='align-text-center m-0'>Select Time</p>
                    <Row className='d-flex justify-content-center'>
                      {slotTime.map((item, index) => {
                        return (
                          <Col
                            key={index}
                            className={
                              'col-3 p-2 m-3 fw-bold ' + (activeTime === item ? 'active' : 'times')
                            }
                          >
                            <TimeCard time={item} setActiveTime={setActiveTime} />
                          </Col>
                        )
                      })}
                    </Row>
                  </div>
                </Modal.Footer>
              </Modal>
            </Row>
          </Col>
        </Row>
        <Row className='cartFooter'>
          <Col className='col-1 text-muted m-auto d-flex justify-content-center fs-5 cartFontawesome'>
            <i className='fa-solid fa-cart-arrow-down' />
          </Col>
          <Col className='col-5 text-center text-muted align-self-center d-flex justify-content-center'>
            Total: ₹ {totalAmount(cartItem) - totalDiscount(cartItem)}
          </Col>
          <Col className='col-6 text-center'>
            <Row className=' p-2'>
              <Button onClick={handleShow3}>{IssueData.constData.cartPreview.bookNow}</Button>
            </Row>
          </Col>
        </Row>
      </Container>
    </>
  )
}

export default CartPreview
