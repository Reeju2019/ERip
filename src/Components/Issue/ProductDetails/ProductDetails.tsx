import { Button, Col, Container, Modal, Row } from 'react-bootstrap'
import RepairServiceCard from './RepairServiceCard'
import './ProductDetails.css'
import ProductData from '../../../Data/Product.mock.json'
import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import Slider from 'react-slick'
import moment from 'moment'
import timeSlotData from '../../../Data/Timeslot.data.json'

// interface SingleService {
//   serviceImage: string
//   service_name: string
//   price: number
//   off: number
//   repairTime: string
//   warranty: string
//   repairType: string
// }

interface Service {
  id: string
  model_name: string
  image: string
  service: {
    serviceImage: string
    service_name: string
    price: number
    off: number
    repairTime: string
    warranty: string
    repairType: string
  }[]
}

const ProductDetails: React.FunctionComponent = () => {
  const { deviceId, brandId, modelId } = useParams()
  // console.log(deviceId, brandId, modelId)

  const [modelData, setModelData] = useState<Service>()
  // const [cartItem, setCartItem] = useState<SingleService[]>([])
  const [show1, setShow1] = useState(false)
  const handleClose1 = () => setShow1(false)
  const handleShow1 = () => setShow1(true)
  const [show2, setShow2] = useState(false)
  const handleClose2 = () => setShow2(false)
  const handleShow2 = () => setShow2(true)
  const [show3, setShow3] = useState(false)
  const handleClose3 = () => setShow3(false)
  const handleShow3 = () => setShow3(true)
  // const [active, setActive] = useState(null)
 const [filterTime, setFilterTime] = useState([])

  useEffect(() => {
    localStorage.setItem('modelId', JSON.stringify(modelId))
    localStorage.setItem('brandId', JSON.stringify(brandId))
    localStorage.setItem('deviceId', JSON.stringify(deviceId))
    ProductData.device.map((deviceItem) => {
      if (deviceItem.device_name === deviceId) {
        deviceItem.brand.map((brandItem) => {
          if (brandItem.brand_name === brandId) {
            brandItem.model.map((modelItem) => {
              if (modelItem.id === modelId) {
                setModelData(modelItem)
              }
            })
          }
        })
      }
    })
    
    // @ filter time slot
    // const temp:any=timeSlotData.timeSlot.filter(item=>{
    //   return item.start_time_hour>moment().get('hour')
    // })

    // setFilterTime(temp)



  }, [])
  // console.log(cart)

  const Slot = () => {
    const days: any = []
    const dateStart = moment()
    const dateEnd = moment().add(7, 'days')
    
    while (dateEnd.diff(dateStart, 'days') >= 0) {
      days.push(dateStart.format('dddd Do') && dateStart.calendar())
      dateStart.add(1, 'days')
    }
    return days
  }

  const Week = () => {
    const weekDays:any =[]
    const weekStart = moment()
    const weekEnd = moment().add(7, 'days')
    
    while (weekEnd.diff(weekStart, 'days') >= 0) {
      weekDays.push(weekStart.calendar({
        sameDay: '[Today]',
        nextDay: '[Tomorrow]',
        sameElse: 'dddd'
    }))
      weekStart.add(1, 'days')
    }
    return weekDays
  }
  console.log(Week(), "nbnb");
  
  


  const settings = {
    accessibility: true,
    adaptiveHeight: false,
    arrows: true,
    autoplay: false,
    autoplaySpeed: 3000,
    centerMode: false,
    centerPadding: '50px',
    cssEase: 'ease',
    dots: false,
    dotsClass: 'slick-dots',
    draggable: true,
    easing: 'linear',
    edgeFriction: 0.15,
    fade: false,
    focusOnSelect: false,
    focusOnChange: false,
    infinite: false,
    initialSlide: 0,
    mobileFirst: false,
    nextArrow: (
      <svg
        xmlns='http://www.w3.org/2000/svg'
        width='16'
        height='16'
        fill='currentColor'
        className='bi bi-arrow-right-short'
        viewBox='0 0 16 16'
      >
        <path
          fillRule='evenodd'
          d='M4 8a.5.5 0 0 1 .5-.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5A.5.5 0 0 1 4 8z'
        />
      </svg>
    ),
    pauseOnDotsHover: false,
    pauseOnFocus: true,
    pauseOnHover: true,
    prevArrow: (
      <svg
        xmlns='http://www.w3.org/2000/svg'
        width='16'
        height='16'
        fill='currentColor'
        className='bi bi-arrow-left-short'
        viewBox='0 0 16 16'
      >
        <path
          fillRule='evenodd'
          d='M12 8a.5.5 0 0 1-.5.5H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5a.5.5 0 0 1 .5.5z'
        />
      </svg>
    ),
    respondTo: 'window',
    speed: 500,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
    rows: 1,
    rtl: false,
    slide: '',
    slidesPerRow: 1,
    slidesToScroll: 1,
    slidesToShow: 4,
    swipe: true,
    swipeToSlide: false,
    touchMove: true,
    touchThreshold: 5,
    useCSS: true,
    useTransform: true,
    variableWidth: false,
    vertical: false,
    verticalSwiping: false,
    waitForAnimate: true,
    zIndex: 1000,
  }

  console.log(moment().format('LT'))

  // useEffect(() => {
  //   Slot()
  // }, [])
console.log(Slot(),"bb");

  // let a = moment().get('hour')
  // console.log(a);


  // console.log(timeSlot.timeSlot[0].id)

  useEffect(() => {
    const filterTimeSlot:any = timeSlotData.timeSlot.filter(function (item) {
      return  item?.start_time_hour > moment().get('hour')
      
    })
    setFilterTime(filterTimeSlot)
 
  }, [])

 const handleSelection =(event:any)=>{
  if (event?.target.name === 'time') {
    console.log(event.target.value);
    
  }
  
 }





  return (
    <>
      <section className='bg-grey'>
        <Container>
          <Row>
            <Col className='col-6 bg-white m-1'>
              <Row>
                <div className='d-flex flex-row'>
                  <div>
                    <img src={modelData?.image} className='prodImg' alt='' />
                  </div>
                  <h4 className='align-self-center fs-5'>{modelData?.model_name}</h4>
                </div>
              </Row>
              <Row>
                <p className='m-0 mx-2 fs-5'>Select your Repair Services</p>
              </Row>
              <Row>
                <div className='d-flex flex-wrap'>
                  {modelData?.service?.map((item, key) => {
                    return (
                      <div key={key} className='col-6 mb-2 mt-0'>
                        <hr />
                        <RepairServiceCard service={item} />
                      </div>
                    )
                  })}
                </div>
              </Row>
            </Col>
            <Col className='col-5 m-2'>
              <Row>
                <Col className='bg-white mx-3 mt-0 p-5'>
                  <h5>Cart Preview</h5>
                  <hr />
                  <div>
                    <div className='my-3 d-flex flex-row justify-content-between text-muted lh-1'>
                      <p className='m-0'>iPhone 13 Pro Max - OEM Screen</p>
                      <p className='m-0 text-primary'>₹ 32999</p>
                    </div>
                    <hr className='m-0' />
                    <div className='my-3 d-flex flex-row justify-content-between text-muted lh-1'>
                      <p className='m-0 '>iPhone 13 Pro Max - Touch & Glass</p>
                      <p className='m-0 text-primary'>₹ 19999</p>
                    </div>
                    <hr className='m-0' />
                    <div className='my-3 d-flex flex-row justify-content-between text-muted lh-1'>
                      <p className='m-0'>iPhone 13 Pro Max - Back Glass</p>
                      <p className='m-0 text-primary'>₹ 8999</p>
                    </div>
                    <hr className='m-0' />
                  </div>
                  <div>
                    <div className='my-3 d-flex flex-row justify-content-between lh-1'>
                      <p className='m-0'>
                        You saved <span className='text-success'>(33%)</span>
                      </p>
                      <p className='m-0 text-primary'>₹ 30998</p>
                    </div>
                    <hr className='m-0' />
                  </div>
                  <div>
                    <div className='my-3 d-flex flex-row justify-content-between lh-1'>
                      <p className='m-0'>Total Amount</p>
                      <p className='m-0 text-primary'>₹ 61997</p>
                    </div>
                    <hr className='m-0' />
                  </div>

                  <Row className='px-2 mt-5'>
                    <Button onClick={handleShow3}>Book Now</Button>
                  </Row>
                </Col>
              </Row>
              <Row>
                <Modal show={show3} onHide={handleClose3} keyboard={false}>
                  <Modal.Header>
                    <Modal.Title>Schedule Appointment</Modal.Title>
                  </Modal.Header>
                  <Modal.Body>
                    <Container>
                      <div>
                        <p>Select Date</p>
                        <Slider {...settings}>
                          {Slot().map((e: any, index: any) => (
                            <div onClick={handleSelection}
                              className='card date_picker slot_card'
                              style={{ width: '18rem' }}
                              key={index}
                            >
                              <div className='card-body form-check slot_card_body'>
                                <label className='form-check-label'>
                                  <input
                                    className='form-check-input radio_slot'
                                    type='radio'
                                    name='booking-date'
                                    value={e.index}
                                  />
                                  <span>{e}</span>
                                </label>
                              </div>
                            </div>
                          ))}
                        </Slider>
                      </div>
                      <div className='row'>
                        {filterTime &&  filterTime.map((e: any, id: number) => {
                          return (
                            <button className='d-flex col-4 col-sm-4 g-0' key={id} onClick={handleSelection}>
                              <label>
                                {e.start_time} - {e.end_time}
                                <input type="radio" value={e.id} name="time" />
                              </label>
                            </button>
                          )
                        })}
                      </div>
                    </Container>
                  </Modal.Body>
                </Modal>
                <div className='bg-white mx-3 mt-4 p-5'>
                  <h5>Check Available Offers</h5>
                  <hr />
                  <div className='offerBorder my-3' onClick={handleShow1}>
                    <div className='d-flex flex-row m-2 w-100'>
                      <img src='/static/singleModel/coupon.png' alt='coupon' />
                      <h6 className='couponText'>Book now. Pay after service</h6>
                    </div>
                    <i className='fa-solid fa-angle-right fontAwesome' />
                  </div>
                  <Modal show={show1} onHide={handleClose1} backdrop='static' keyboard={false}>
                    <Modal.Header closeButton>
                      <Modal.Title>Modal title</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                      I will not close if you click outside me. Dont even try to press escape key.
                    </Modal.Body>
                  </Modal>
                  <div className='offerBorder my-3' onClick={handleShow2}>
                    <div className='d-flex flex-row m-2 w-100'>
                      <img src='/static/singleModel/coupon.png' alt='coupon' />
                      <h6 className='couponText'>Get upto 35% off on your first service</h6>
                    </div>
                    <i className='fa-solid fa-angle-right fontAwesome' />
                  </div>
                  <Modal show={show2} onHide={handleClose2} backdrop='static' keyboard={false}>
                    <Modal.Header closeButton>
                      <Modal.Title>Modal title</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>I will not close if you click outside me. second part</Modal.Body>
                  </Modal>
                  <div className='d-flex flex-row justify-content-between'>
                    <div>
                      <input type='text' placeholder='Enter Coupon code' className='p-1' />
                    </div>
                    <div>
                      <Button className='applyBtn'>Apply</Button>
                    </div>
                  </div>
                </div>
              </Row>
            </Col>
          </Row>
        </Container>
      </section>
    </>
  )
}

export default ProductDetails
