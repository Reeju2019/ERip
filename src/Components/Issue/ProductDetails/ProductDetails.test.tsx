import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ProductDetails from './ProductDetails'

test('product details', () => {
  render(<ProductDetails />)
})
describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<ProductDetails />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<ProductDetails />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
