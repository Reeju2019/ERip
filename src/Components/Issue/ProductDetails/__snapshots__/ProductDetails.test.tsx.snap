// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<section
  className="bg-grey pb-5"
  data-testid="custom-element"
>
  <div
    className="container"
  >
    <div
      className="d-flex justify-content-between row"
    >
      <div
        className="w-100 bg-white m-1 col"
      >
        <div
          className="row"
        >
          <div
            className="d-flex flex-row"
          >
            <div>
              <img
                alt=""
                className="prodImg"
              />
            </div>
            <h4
              className="align-self-center fs-5"
            />
          </div>
        </div>
        <div
          className="row"
        >
          <p
            className="m-0 mx-2 fs-5"
          >
            <b>
              Select your Repair Services
            </b>
          </p>
        </div>
        <hr />
        <div
          className="row"
        >
          <div
            className="d-flex flex-wrap"
          />
        </div>
      </div>
      <div
        className="col-5 m-1 col"
      >
        <div
          className="w-100 mx-1 container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="bg-white mt-0 p-5 col"
            >
              <h5>
                Cart Preview
              </h5>
              <hr />
              <div />
              <div>
                <div
                  className="my-3 d-flex flex-row justify-content-between lh-1"
                >
                  <p
                    className="m-0"
                  >
                    You Saved
                    <span
                      className="text-success"
                    >
                       
                      (
                      0
                    </span>
                    <span
                      className="text-success"
                    >
                      %)
                    </span>
                  </p>
                  <p
                    className="m-0 text-primary"
                  >
                    ₹ 
                    0
                  </p>
                </div>
                <hr
                  className="m-0"
                />
              </div>
              <div>
                <div
                  className="my-3 d-flex flex-row justify-content-between lh-1"
                >
                  <p
                    className="m-0"
                  >
                    Total Amount
                  </p>
                  <p
                    className="m-0 text-primary"
                  >
                    ₹ 
                    0
                  </p>
                </div>
                <hr
                  className="m-0"
                />
              </div>
              <div
                className="px-2 mt-5 row"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Book Now
                </button>
              </div>
            </div>
          </div>
          <div
            className="cartFooter row"
          >
            <div
              className="col-1 text-muted m-auto d-flex justify-content-center fs-5 cartFontawesome col"
            >
              <i
                className="fa-solid fa-cart-arrow-down"
              />
            </div>
            <div
              className="col-5 text-center text-muted align-self-center d-flex justify-content-center col"
            >
              Total: ₹ 
              0
            </div>
            <div
              className="col-6 text-center col"
            >
              <div
                className=" p-2 row"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Book Now
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="prodRow row"
        >
          <div
            className="bg-white mx-3 mt-4 p-5"
          >
            <h5>
              Check Available Offers
            </h5>
            <hr />
            <div
              className="offerBorder my-3"
              onClick={[Function]}
            >
              <div
                className="d-flex flex-row m-2 w-100"
              >
                <img
                  alt="coupon"
                  src="/static/singleModel/coupon.png"
                />
                <h6
                  className="couponText"
                >
                  Book now. Pay after service
                </h6>
              </div>
              <i
                className="fa-solid fa-angle-right fontAwesome"
              />
            </div>
            <div
              className="offerBorder my-3"
              onClick={[Function]}
            >
              <div
                className="d-flex flex-row m-2 w-100"
              >
                <img
                  alt="coupon"
                  src="/static/singleModel/coupon.png"
                />
                <h6
                  className="couponText"
                >
                  Get upto 35% off on your first service
                </h6>
              </div>
              <i
                className="fa-solid fa-angle-right fontAwesome"
              />
            </div>
            <div
              className="d-flex flex-row justify-content-between"
            >
              <div>
                <input
                  className="p-1"
                  placeholder="Enter Coupon code"
                  type="text"
                />
              </div>
              <div>
                <button
                  className="applyBtn btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  Apply
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
